src/datagrid/DataGrid.tsx

```diff
@@ -583,6 +614,13 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
         }
     };
 
+    // Function to handle CustomPageSize change in input box
+    private handleCustomPageSizeChange = () => {
+        const pageSize = this.pageSizeRef.current && parseInt(this.pageSizeRef.current.value);
+        if (pageSize && pageSize <= MAX_PAGE_SIZE) {
```

Comment:
We need to show an error to the user that the max limit for custom page size is 1000. or whatever the user has provided.
or if the user enters wrong value then we need to set it to max limit